bindings/python/verovio.i

```diff
@@ -27,142 +29,140 @@
 // Toolkit::Edit
 %feature("shadow") vrv::Toolkit::Edit( const std::string & ) %{
 def edit(toolkit, editor_action):
+    """Edit the MEI data."""
     return $action(toolkit, json.dumps(editor_action))
 %}
 
 // Toolkit::EditInfo
 %feature("shadow") vrv::Toolkit::EditInfo() %{
 def editInfo(toolkit):
+    """Return the editor status."""
     return json.loads($action(toolkit))
 %}
 
 // Toolkit::GetAvailableOptions
 %feature("shadow") vrv::Toolkit::GetAvailableOptions() const %{
-def getAvailableOptions(toolkit):
+def getAvailableOptions(toolkit) -> dict:
+    """Return a dictionary of all the options with their default value."""
     return json.loads($action(toolkit))
 %}
 
 // Toolkit::GetDefaultOptions
 %feature("shadow") vrv::Toolkit::GetDefaultOptions() const %{
-def getDefaultOptions(toolkit):
+def getDefaultOptions(toolkit) -> dict:
+    """Return a dictionary of all the options with their default value."""
     return json.loads($action(toolkit))
 %}
 
 // Toolkit::GetDescriptiveFeatures
 %feature("shadow") vrv::Toolkit::GetDescriptiveFeatures(const std::string &) %{
-def getDescriptiveFeatures(toolkit, options):
+def getDescriptiveFeatures(toolkit, options: dict = {}) -> dict:
+    """Return descriptive features as dictionary."""
     return json.loads($action(toolkit, json.dumps(options)))
 %}
 
 // Toolkit::GetElementAttr
 %feature("shadow") vrv::Toolkit::GetElementAttr(const std::string &) %{
-def getElementAttr(toolkit, xml_id):
+def getElementAttr(toolkit, xml_id: str) -> dict:
+    """Return element attributes as dictionary."""
     return json.loads($action(toolkit, xml_id))
 %}
 
 // Toolkit::GetElementsAtTime
 %feature("shadow") vrv::Toolkit::GetElementsAtTime(int) %{
-def getElementsAtTime(toolkit, millisec):
+def getElementsAtTime(toolkit, millisec: int) -> dict:
+    """Return array of IDs of elements being currently played."""
     return json.loads($action(toolkit, millisec))
 %}
 
 // Toolkit::GetExpansionIdsForElement
 %feature("shadow") vrv::Toolkit::GetExpansionIdsForElement(const std::string &) %{
-def getExpansionIdsForElement(toolkit, xml_id):
+def getExpansionIdsForElement(toolkit, xml_id: str) -> dict:
+    """Return a vector of ID strings of all elements (the notated and the expanded) for a given element."""
     return json.loads($action(toolkit, xml_id))
 %}
 
 // Toolkit::GetMEI
 %feature("shadow") vrv::Toolkit::GetMEI(const std::string & = "") %{
-def getMEI(toolkit, options = None):
-    if options == None:
-        options = {}
+def getMEI(toolkit, options: dict = {}) -> str:
```

Comment:
This is incorrect. Mutable structures should not be used as default arguments.

https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Using equality operators here hurts even more. But I'll change it, thanks.

Reply:
You’re correct, though — it should be `is None` instead of `== None`

Reply:
> it should be is None instead of == None

That was me... ;-)